src/chart/LineChart.tsx

```diff
@@ -1,10 +1,21 @@
-/**
- * @fileOverview Line Chart
- */
-import { generateCategoricalChart } from './generateCategoricalChart';
+import * as React from 'react';
+import { forwardRef } from 'react';
 import { arrayTooltipSearcher } from '../state/optionsSlice';
+import { CartesianChart } from './CartesianChart';
+import { CategoricalChartProps } from './generateCategoricalChart';
+import { TooltipEventType } from '../util/types';
 
-export const LineChart = generateCategoricalChart({
-  chartName: 'LineChart',
-  tooltipPayloadSearcher: arrayTooltipSearcher,
+const allowedTooltipTypes: ReadonlyArray<TooltipEventType> = ['axis'];
+
+export const LineChart = forwardRef<SVGSVGElement, CategoricalChartProps>((props: CategoricalChartProps, ref) => {
```

Comment:
Wonder how dramatically this changes the type. Good thing we're doing it as breaking